Microsoft.Toolkit/Helpers/ObjectStorage/SystemSerializer.cs

```diff
@@ -0,0 +1,46 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Reflection;
+
+namespace Microsoft.Toolkit.Helpers
+{
+    /// <summary>
+    /// A bare-bones serializer which knows how to deal with primitive types and strings only.
+    /// It is recommended for more complex scenarios to implement your own <see cref="IObjectSerializer"/> based on System.Text.Json, Newtonsoft.Json, or DataContractJsonSerializer see https://aka.ms/wct/storagehelper-migration
+    /// </summary>
+    public class SystemSerializer : IObjectSerializer
+    {
+        /// <summary>
+        /// Take a primitive value from storage and return it as the requested type using the <see cref="Convert.ChangeType(object, Type)"/> API.
+        /// </summary>
+        /// <typeparam name="T">Type to convert value to.</typeparam>
+        /// <param name="value">Value from storage to convert.</param>
+        /// <returns>Deserialized value or default value.</returns>
+        public T Deserialize<T>(string value)
+        {
+            var type = typeof(T);
+            var typeInfo = type.GetTypeInfo();
+
+            if (typeInfo.IsPrimitive || type == typeof(string))
+            {
+                return (T)Convert.ChangeType(value, type);
+            }
+
+            throw new NotSupportedException("This serializer can only handle primitive types and strings. Please implement your own IObjectSerializer for more complex scenarios.");
+        }
+
+        /// <summary>
+        /// Returns the value so that it can be serialized directly.
+        /// </summary>
+        /// <typeparam name="T">Type to serialize from.</typeparam>
+        /// <param name="value">Value to serialize.</param>
+        /// <returns>String representation of value.</returns>
+        public string? Serialize<T>(T value)
+        {
+            return value?.ToString();
```

Comment:
Note: this change makes this path less efficient (eg. primitive values will first be formatted to a `string`, then serialized again when saving them with the UWP APIs). If we have decided we don't really care about the new/refactored types having an ideal API shape (they don't anyway), then I think we might consider just reverting this bit to the original. I'm not convinced this type should really ever be used by anyone outside of just the specific built-in usages in the Toolkit anyway (and we shouldn't really recommend anyone using this type themselves either).

Reply:
> I'm not convinced this type should really ever be used by anyone outside of just the specific built-in usages in the Toolkit anyway

Exactly ^. Though these are not otherwise changes I was planning on making, I agree with your earlier points. I think converting values between `string` and `T` is better than `object` and `T` because we can rationalize that the `string` value represents serialized content and is not some other random object. The unfortunate side effect is that SystemSerializer is now less efficient.

@Sergio0694 what do you think? Is there a preferred path in your mind that satisfies these issues?

Reply:
Yeah I agree that fixing the output types to a `string` makes the contract clear instead of just representing some opaque serialization interface. That deoptimization in the system serializer is unfortunate but I guess it might not be that crucial if that will only ever be used to serialize small things like numbers and other primitives at least. Also thankfully `string.ToString()` just returns `this`, so at least we're not making a whole copy of `string`-s being serialized 😄

Not really seeing an easy solution to make this much better at this point given the constraints and deadlines we agreed upon for 7.1, and I don't really want to delay your PR any further either, so I think this is fine for this release 👍